devito/function.py

```diff
@@ -744,6 +748,23 @@ def dt2(self):
 
         return self.diff(_t, _t).as_finite_difference(indt)
 
+    def argument_values(self, alias=None, **kwargs):
+        """
+        Returns a map of argument values after evaluating user input.
+
+        :param kwargs: Dictionary of user-provided argument overrides.
+        :param alias: (Optional) name under which to store values.
+        """
+        # Check if data has the right dimension
+        if self.name in kwargs:
+            new = kwargs.get(self.name)
+            if isinstance(new, TimeFunction) and new.save != self.save:
```

Comment:
Note for the future: we should error out if ``new`` is not a TimeFunction (with a different error msg)